surefire-booter/src/main/java/org/apache/maven/surefire/booter/ForkedBooter.java

```diff
@@ -129,6 +134,34 @@ private void setupBooter( String tmpDir, String dumpFileName, String surefirePro
 
         flushEventChannelOnExit();
 
+        final Long outputFlushInterval = providerConfiguration.getOutputFlushInterval();
+        if ( outputFlushInterval != null && outputFlushInterval > 0
+            && LegacyMasterProcessChannelEncoder.class.isInstance( eventChannel ) )
```

Comment:
instanceof smells a little.
do we have any more object oriented friendly way ?

Reply:
For the same reason than the default is 0, I didn't want to modify the API there for this (non-)feature so instanceof was the best compromise IMHO even if not as elegant as adding 2 methods.

Reply:
I see.
The API is still unstable, maybe it is better to change it now.
I am fine with keeping this trick here as an hotfix.
but we should create a JIRA at least to not forget